### PR TITLE
Build with numpy>=1.25; add CI

### LINKED ISSUE #24: Add a CI workflow
There's no unit tests for this repo, but a CI workflow could test that the docs build, that flake8 checks pass, etc.

### PR DESCRIPTION
**Pull request by jdavies-st**
Resolves #24

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,69 @@
...
+name = "stsci.imagestats"
+dynamic = ["version"]
+description = "Compute sigma-clipped statistics on data arrays"
+readme = "README.rst"
+authors = [
+    {name = "Warren Hack", email = "help@stsci.edu"},
+    {name = "Christopher Hanley", email = "help@stsci.edu"},
+]
+license = { text = "BSD-3-Clause" }
+requires-python = ">=3.8"
```
Comment (mcara) [inline pyproject.toml:11]: Is `3.8` really _required_ for this code to run?

File: pyproject.toml [hunk 2]
```diff
@@ -1,10 +1,69 @@
...
+    "Programming Language :: C",
+    "Programming Language :: Cython",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering :: Astronomy",
+    "Topic :: Scientific/Engineering :: Physics",
+    "Development Status :: 7 - Inactive",
+]
+dependencies = [
+    "numpy>=1.20",
```
Comment (mcara) [inline pyproject.toml:25]: Is this restriction necessary?

File: pyproject.toml [hunk 3]
```diff
@@ -1,10 +1,69 @@
...
+
+
 [build-system]
 requires = [
     "setuptools>=38.2.5",
-    "setuptools_scm",
+    "setuptools_scm>=6.2",
     "wheel",
-    "oldest-supported-numpy",
+    "numpy>=1.25,<2",
```
Comment (mcara) [inline pyproject.toml:51]: Why there is an upper limit?

**Comment by mcara** [comment]
Why is "Build with numpy>=1.25" _needed_?

**Review by mcara — APPROVED** [review]
I will merge this as is and remove unnecessary restrictions in a different PR.

File: pyproject.toml (hunk 3 above)
Comment (pllim) [reply pyproject.toml:56]: numpy 2.0 has ABI changes that are not backward compatible. But that said, if you know your package is compatible with numpy 2.x ABI, you can unpin it here. I am sure not sure how we're supposed to package for both numpy 1.x and 2.x . Keep an eye on astropy releases.

File: stsci/imagestats/test/test_imagestats.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import numpy as np
+
+from stsci.imagestats import ImageStats
+
+
+def test_imagestats():
+    mean = 1.854
+    stddev = 0.333
+    a = np.random.normal(loc=mean, scale=stddev, size=(100, 200))
```
Comment (pllim) [inline stsci/imagestats/test/test_imagestats.py:9]: Didn't even set a seed. You're living dangerously.

File: tox.ini [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    numpy123: numpy==1.23.*
+
+# The following indicates which [project.optional-dependencies] from pyproject.toml will be installed
+extras =
+    test: test
+
+commands =
+    pip freeze
+    !cov: pytest --pyargs stsci.imagestats {posargs}
+    cov: pytest --pyargs stsci.imagestats --cov stsci.imagestats --cov-config={toxinidir}/pyproject.toml --cov-report xml:{toxinidir}/coverage.xml {posargs}
```
Comment (pllim) [inline tox.ini:40]: cov is not used at all in the actual workflow.

**Comment by jdavies-st** [comment]
> Why is "Build with numpy>=1.25" _needed_?

Because previous versions of numpy have produce code that is not ABI compatible and require the use of oldest-supported-numpy.  WIth 1.25 and later, this is no longer the case, and oldest-supported-numpy is no longer needed.  That's the whole point of this PR (along with adding CI).  See more here

https://numpy.org/doc/stable/dev/depending_on_numpy.html#build-time-dependency

**Pull request merged**